EIPS/eip-4910.md

```diff
@@ -0,0 +1,979 @@
+
+#### Listing NFTs for Sale and removing a Listing
+
+Authorized user addresses can list NFTs for sale for non-exchange mediated NFT purchases.
+
+#### Payment Function from Buyer to Seller
+
+To avoid royalty circumvention, a buyer will always pay the NFT contract directly and not the seller. The seller is paid through the royalty distribution and can later request a payout.
+
+The payment process depends on whether the payment is received in ETH or an ERC 20 token:
+* ERC-20 Token
+    1. The Buyer must `approve` the NFT contract for the purchase price, `payment` for the selected payment token (ERC20 contract address).
+    2. For an ERC20 payment token, the Buyer must then call the `executePayment` in the NFT contract -- the ERC20 is not directly involved.
+* For a non-ERC20 payment, the Buyer must send a protocol token (ETH) to the NFT contract, and is required to send `msg.data` encoded as an array of purchased NFTs `uint256[] tokenId`.
+
+#### Modified NFT Transfer Function including required Trade data to allocate Royalties
+
+The input parameters must satisfy several requirements for the NFT to be transferred AFTER the royalties have been properly distributed. Furthermore, the ability to transfer more than one token at a time is also considered.
+
+The proposal defines:
+* Input parameter validation
+* Payment Parameter Validation
+* Distributing Royalties
+* Update Royalty Account ownership with payout
+* Transferring Ownership of the NFT 
+* Removing the Payment entry in `registeredPayment` after successful transfer
+
+##### Distributing Royalties
+
+The approach to distributing royalties is to break down the hierarchical structure of interconnected Royalty Accounts into layers and then process one layer at time, where each relationship between a token and its ancestor is utilized to traverse the Royalty Account chain until the root ancestor and associated RA is reached.
+
+#### Paying out Royalties to the NFT Owner -- `from` address in `safeTransferFrom` Function
+
+This is the final part of the proposal.
+
+There are two versions of the payout function -- a `public` function and an `internal` function.
+
+The public function has the following interface:
+```
+function royaltyPayOut (uint256 tokenId, address RAsubaccount, address payable payoutAccount, uint256 amount) public virtual nonReentrant returns (bool)
+```
+
+where we only need the `tokenId`, the RA Sub Account address, `_RAsubaccount` which is the `owner`, and the amount to be paid out, `_amount`. Note that the function has `nonReentrant` modifier protection, because funds are being payed out.
+
+#### Sending a Payout Payment
+
+The following steps need to be taken:
+* find the RA Sub Account based on `RAaccount` and the `subaccountPos` and extract the balance
+* extract `tokentype` from the Sub Account
+* based on the token type, send the payout payment (not exceeding the available balance)
+
+### Data Structures
+
+#### Rationale: Royalty Account and Royalty Sub Accounts
```

Comment:
Again, sorry for this taking a couple rounds to convey. I'm still new at this EIP editing thing :sweat_smile: 

The current structure is:

```markdown
# Specification

## Data Structures

### Rationale: Royalty Account and Royalty Sub Accounts

### Specification: Royalty Account and Royalty Sub Accounts
```

It needs to be:

```markdown
# Specification

## Data Structures

### Royalty Account and Royalty Sub Accounts

# Rationale

## Data Structures

### Royalty Account and Royalty Sub Accounts
```

(alternatively, if the rationale is small enough, you can sneak it in the specification section without a header :smirk:)

Reply:
@SamWilsn  No issue whatsoever. So sorry for the oversight. I overlooked that one area. I snuck all of the rationales in at the beginning of each specification section since they are generally very short, as you suggested.

Reply:
Minor correction, the top level sections should be `##` not `#`.  Don't ask me why, it is one of those things where the original reasoning has been lost to time and we continue doing it because consistency is more important than doing it more correctly.

Reply:
I doublechecked @SamWilsn and @MicahZoltu and the top-level is `##` in the doc.